### PR TITLE
Add entity reference doc.

### PR DESCRIPTION
**Pull request by hershey5045**
Signed-off-by: hershey5045 <43485729+hershey5045@users.noreply.github.com>

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by galibzon — APPROVED** [review]

File: content/docs/user-guide/components/reference/atom/entity-reference.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: 'Open 3D Engine (O3DE) Entity Reference component reference.'
 toc: true
 ---
 
-The **Entity Reference** component allows you to provide an entity with references to other entities
+The **Entity Reference** component allows tracking one or more entity IDs that can be conveniently accessed through script or C++ code during editor mode.
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/entity-reference.md:8]: Switch to active voice - answers _who_ is doing the tracking and accessing of entity IDs. Users have to set this up and do this. 
```suggestion
The **Entity Reference** component allows you to track one or more entity IDs that you can conveniently access through script or C++ code during editor mode.
```

File: content/docs/user-guide/components/reference/atom/entity-reference.md [hunk 2]
```diff
@@ -14,6 +14,25 @@ The **Entity Reference** component allows you to provide an entity with referenc
...
 ## Properties
 
-{{< todo issue=https://github.com/o3de/o3de.org/issues/717 >}}
-Add properties.
-{{< /todo >}}
+![Entity Reference Component UI](/images/user-guide/components/reference/atom/entity-reference-ui.png)
+
+| Property | Description |
+| - | - |
+| **EntityIdReferences** | Contains a list of entities whose ids will be tracked |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/entity-reference.md:21]:
```suggestion
| **EntityIdReferences** | Contains a list of entities whose IDs will be tracked. |
```

File: content/docs/user-guide/components/reference/atom/entity-reference.md [hunk 3]
```diff
@@ -14,6 +14,25 @@ The **Entity Reference** component allows you to provide an entity with referenc
...
+
+## Usage ##
+
+During editor mode, call the EntityReferenceRequestBus to retrieve the list of entity being referenced by the component.
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/entity-reference.md:25]:
```suggestion
During editor mode, call the `EntityReferenceRequestBus` to retrieve the list of entities being referenced by the component.
```

File: content/docs/user-guide/components/reference/atom/entity-reference.md [hunk 4]
```diff
@@ -14,6 +14,25 @@ The **Entity Reference** component allows you to provide an entity with referenc
...
+This example uses the editor's python console to retrieve a list of entity ids.
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/entity-reference.md:26]:
```suggestion
This example uses the O3DE's Python Console to retrieve a list of entity IDs.
```

File: content/docs/user-guide/components/reference/atom/entity-reference.md [hunk 5]
```diff
@@ -14,6 +14,25 @@ The **Entity Reference** component allows you to provide an entity with referenc
...
+```
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/entity-reference.md:27]: Add python syntax highlighting by adding "python" after "```"
Will look like this: 
image

As opposed to this:
image

**Review by chanmosq — APPROVED** [review]

File: content/docs/user-guide/components/reference/atom/entity-reference.md (hunk 5 above)
Comment (hershey5045) [reply content/docs/user-guide/components/reference/atom/entity-reference.md]: Awesome suggestion, done!

**Comment by chanmosq** [comment]
Looks great, thank you!

**Pull request merged**